### PR TITLE
Add support for Kubernetes 1.18

### LINKED ISSUE #140: Kubernetes 1.18
## What

Update dependencies, manifests, and tests for Kubernetes 1.18.

## How

Since support for k8s 1.15 will be dropped, [e2e/legacy](https://github.com/cybozu-go/topolvm/blob/master/e2e/legacy) and the following checks can be deleted.

- https://github.com/topolvm/topolvm/blob/73aa20d57d582aba4c041d23e7d56210c2796af9/e2e/e2e_test.go#L725-L731
- https://github.com/topolvm/topolvm/blob/73aa20d57d582aba4c041d23e7d56210c2796af9/e2e/e2e_test.go#L794-L800
- https://github.com/topolvm/topolvm/blob/73aa20d57d582aba4c041d23e7d56210c2796af9/e2e/e2e_test.go#L966-L972

## Checklist

- [x] Finish implementation of the issue
- [x] Test all functions
- [x] Have enough logs to trace activities
- [x] Notify developers of necessary actions

### PR DESCRIPTION
**Pull request by ymmt2005**
This PR adds support for Kubernetes 1.18 and removes that for 1.15.
Notables changes are:

- CSI sidecars
    - external-provisioner 1.6.0
    - external-attacher 2.2.0
    - external-resizer 1.0.0
    - node-driver-registrar 2.0.1
    - livenessprobe 2.1.0
- kubebuilder 2.3.1
- controller-runtime 0.6.3
- controller-tools 0.4.0
- Update gRPC and protobuf libraries
- Replace cfssl with a tiny Go tool for certificate generation
- Replace the deprecated `golint` with `staticcheck`
    - fixes errors reported by `staticcheck`

Resolves #140

### CODE REVIEW

File: e2e/Makefile [hunk 1]
```diff
@@ -24,13 +24,13 @@ export TEST_KUBERNETES_VERSION
 KUSTOMIZE_DIR=e2ebase
 
 ifeq ($(TEST_KUBERNETES_VERSION),1.18)
-KUBERNETES_VERSION=1.18.9
+KUBERNETES_VERSION=1.18.8
```
Comment (llamerada-jp) [inline e2e/Makefile:27]: Why didn't you use kubernetes 1.18.9?

File: deploy/README.md [hunk 2]
```diff
@@ -12,8 +12,7 @@ Each of these steps are shown in depth in the following sections:
 1. Add `topolvm.cybozu.com/webhook: ignore` label to system namespaces such as `kube-system`.
 1. Apply remaining manifests for TopoLVM from `deploy/manifests/base` plus overlays as appropriate to your installation.
 1. Configure `kube-scheduler` to use `topolvm-scheduler`. 
-1. Adjust CSIDriver config for Kubernetes 1.15 and earlier
-1. Prepare StorageClasses for TopoLVM.
+2. Prepare StorageClasses for TopoLVM.
```
Comment (llamerada-jp) [inline deploy/README.md:15]:
```suggestion
1. Prepare StorageClasses for TopoLVM.
```

File: Makefile [hunk 3]
```diff
@@ -1,56 +1,59 @@
+## Dependency versions
+
 CSI_VERSION=1.1.0
-PROTOBUF_VERSION=3.11.4
+K8S_VERSION=1.18.9
+KUBEBUILDER_VERSION = 2.3.1
+KIND_VERSION=0.9.0
+PROTOC_VERSION=3.12.4
```
[the remark this answers is not in the document]
Comment (ymmt2005) [reply Makefile:7]: It's too new.

File: e2e/Makefile (hunk 1 above)
Comment (ymmt2005) [reply e2e/Makefile:27]: there's no such tag: https://hub.docker.com/r/kindest/node/tags

File: deploy/README.md (hunk 2 above)
Comment (ymmt2005) [reply deploy/README.md:15]: It's harmless.

**Review by llamerada-jp — APPROVED** [review]
LGTM

**Review by satoru-takeuchi — APPROVED** [review]

**Pull request merged**